### PR TITLE
Remove PreludeCommon and Prelude into Common

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

File: core/src/Streamly/Internal/Data/Stream/Common.hs [hunk 1]
```diff
@@ -1,39 +1,83 @@
...
+-- |
+-- Module      : Streamly.Internal.Data.Stream.Common
+-- Copyright   : (c) 2017 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- Low level functions using StreamK as the intermediate stream type. These
+-- functions are used in SerialT/AsyncT/AheadT/ParallelT stream modules to
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Stream/Common.hs:13]: Update the comment.

File: core/src/Streamly/Internal/Data/Stream/Common.hs [hunk 2]
```diff
@@ -1,39 +1,83 @@
...
-toStreamS :: (IS_STREAM Monad m) => STREAM m a -> S.Stream m a
-toStreamS = S.fromStreamK . TO_STREAM
+toStreamS :: Monad m => K.Stream m a -> S.Stream m a
+toStreamS = S.fromStreamK
 
 {-# INLINE toStreamD #-}
-toStreamD :: (IS_STREAM Monad m) => STREAM m a -> D.Stream m a
-toStreamD = D.fromStreamK . TO_STREAM
+toStreamD :: Monad m => K.Stream m a -> D.Stream m a
+toStreamD = D.fromStreamK
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Stream/Common.hs:72]: We can remove toStreamS/fromSrteamS/toStreamD, and directly use toStreamK/fromSrteamK?

Comment (adithyaov) [reply core/src/Streamly/Internal/Data/Stream/Common.hs:72]: Yes

**Review by harendra-kumar — APPROVED** [review]
squash and merge.

**Pull request merged**